runtime/balrt.h

```diff
@@ -271,8 +293,28 @@ extern char *_bal_string_alloc(uint64_t lengthInBytes, uint64_t lengthInCodePoin
 
 extern void _bal_array_grow(GC GenericArray *ap, int64_t min_capacity, int shift);
 extern ListPtr _bal_list_construct(ListDescPtr desc, int64_t capacity);
-extern TaggedPtr _bal_list_get(TaggedPtr p, int64_t index);
-extern PanicCode _bal_list_set(TaggedPtr p, int64_t index, TaggedPtr val);
+
+extern TaggedPtr _bal_list_generic_tagged_get(TaggedPtr p, int64_t index);
+extern int64_t _bal_list_generic_int_get(TaggedPtr p, int64_t index);
+double _bal_list_generic_float_get(TaggedPtr p, int64_t index);
+extern PanicCode _bal_list_generic_tagged_set(TaggedPtr p, int64_t index, TaggedPtr val);
+extern PanicCode _bal_list_generic_int_set(TaggedPtr p, int64_t index, int64_t val);
+PanicCode _bal_list_generic_float_set(TaggedPtr p, int64_t index, double val);
+
+extern TaggedPtr _bal_list_int_array_tagged_get(TaggedPtr p, int64_t index);
+extern int64_t _bal_list_int_array_int_get(TaggedPtr p, int64_t index);
+extern float (*const _bal_list_int_array_float_get)(TaggedPtr p, int64_t index); // Reads can't fail. Will not be called.
```

Comment:
Don't see the point of this.

Reply:
In `init.bal` I blindly create `FunctionDecl` for each combination, so currently I rely on this symbol existing. Alternatively, I can put a `nullptr` in `init.bal` when `ty != "tagged" && ty !=  repr.kind`.

Reply:
The problem with what you have is that the types are bogus. When you refer to the symbol _bal_list_int_array_int_get in LLVM the type will be a pointer to a function. When you refer to the symbol _bal_list_int_array_float_get, you will get a pointer to a pointer to a function.